app/assets/javascripts/index_map.js

```diff
@@ -12,12 +12,19 @@ function initMap() {
 
   var unit_markers = [];
   var infowindows = [];
-  var unit = {};
+  var unit = [];
   var window_content;
+  var heatMapPoint, heatMapData = [];
 
   for (var i in gon.rental_units) {
-    unit = {lat: gon.rental_units[i].latitude,
-                     lng: gon.rental_units[i].longitude};
+    unit[i] = new google.maps.LatLng(gon.rental_units[i].latitude,
```

Comment:
Identifier 'rental_units' is not in camel case.<br>'google' is not defined.<br>'gon' is not defined.